audbackend/core/backend/artifactory.py

```diff
@@ -109,45 +69,106 @@ def _download(
 class Artifactory(Base):
     r"""Backend for Artifactory.
 
-    Looks for the two environment variables
-    ``ARTIFACTORY_USERNAME`` and
-    ``ARTIFACTORY_API_KEY``.
-    Otherwise,
-    tries to extract missing values
-    from a global `config file`_.
-    The default path of the config file
-    (:file:`~/.artifactory_python.cfg`)
-    can be overwritten with the environment variable
-    ``ARTIFACTORY_CONFIG_FILE``.
-    If no config file exists
-    or if it does not contain an
-    entry for the ``host``,
-    the username is set to ``'anonymous'``
-    and the API key to an empty string.
-    In that case the ``host``
-    should support anonymous access.
-
     Args:
         host: host address
         repository: repository name
-
-    .. _`config file`: https://devopshq.github.io/artifactory/#global-configuration-file
+        auth: username, password / API key / access token tuple.
+            If ``None``,
+            it requests it by calling :meth:`authentication`
 
     """  # noqa: E501
 
     def __init__(
         self,
-        host,
-        repository,
+        host: str,
+        repository: str,
+        *,
+        auth: typing.Tuple = None,
     ):
-        super().__init__(host, repository)
+        super().__init__(host, repository, auth=auth)
+
+        if auth is None:
+            self.auth = self.authentication(host)
+
+        # Store ArtifactoryPath object to the repository,
+        # when opening the backend.
+        self._repo = None
+
+    @classmethod
+    def authentication(cls, host: str) -> typing.Tuple[str, str]:
+        """Username and password/access token for given host.
+
+        Returns a username
+        and password / API key / access token,
+        which can be used to authenticate
+        with an Artifactory server.
+
+        Note, API keys are deprecated
+        and will no longer work
+        with newer versions of Artifactory.
+
+        To get the username,
+        password/access token combination,
+        the function looks first
+        for the two environment variables
+        ``ARTIFACTORY_USERNAME`` and
+        ``ARTIFACTORY_API_KEY``.
```

Comment:
Do we also use the variable `ARTIFACTORY_API_KEY` for access tokens?

Reply:
Yes, it's all very messy with names for authentication in Artifactory.
They will remove support for API keys and have it replaced with access tokens, which are very similar but automatically expire after some time.

Do not make it too complicated I decided for now to not add another environment variable. But I agree that it is unfortunate that the name will not match (API key != access token). Would you propose to add `ARTIFACTORY_ACCESS_TOKEN` and then simply first look for that, and if empty/missing look for `ARTIFACTORY_API_KEY` next?

Reply:
No, I'm also not convinced by their new approach. I as an admin have to create new access tokens and update them here on Github.

At the moment our tests here are failing as I still have a bug in the code, which I introduced when renaming `auth` to `authenticaton`. I'm working to solve this at the moment.

---

In this pull request, I would then stay with `ARTIFACTORY_API_KEY`, but created https://github.com/audeering/audbackend/issues/219 to solve this in the future.

Reply:
I found the error, and have fixed it.